network/suite.go

```diff
@@ -63,3 +83,94 @@ func (s *TestSuite) CreateTxBytes(fees sdk.Coin, gas uint64, msgs []sdk.Msg) ([]
 	bz, err := val.ClientCtx.TxConfig.TxEncoder()(builder.GetTx())
 	return bz, err
 }
+
+func (s *TestSuite) GetAccountI(acc account.Account) (sdk.AccountI, error) {
+	cc, closeFn, err := s.GetGRPC()
+	if err != nil {
+		return nil, err
+	}
+	defer closeFn()
+
+	authClient := authtypes.NewQueryClient(cc)
+
+	resp, err := authClient.Account(context.Background(), &authtypes.QueryAccountRequest{Address: acc.Address().String()})
+	if err != nil {
+		return nil, err
+	}
+
+	var accI sdk.AccountI
+	if err := authCodec.UnpackAny(resp.Account, &accI); err != nil {
+		return nil, err
+	}
+
+	return accI, err
+}
+
+func (s *TestSuite) GetCometClient() (*cmthttp.HTTP, error) {
+	return cmthttp.New(s.Network.Validators[0].RPCAddress, "/websocket")
+}
+
+// CreateTx creates and signs a transaction, from the given messages
+func (s *TestSuite) CreateTx(ctx context.Context, acc account.Account, gasLimit, fee, timeoutHeight uint64, chainID string, msgs ...sdk.Msg) ([]byte, error) {
+	accI, err := s.GetAccountI(acc)
+	if err != nil {
+		return nil, err
+	}
+
+	txConfig := s.Network.Validators[0].ClientCtx.TxConfig
+
+	txFactory := clienttx.Factory{}.
+		WithChainID(chainID).
+		WithTxConfig(txConfig).
+		WithSignMode(signing.SignMode_SIGN_MODE_DIRECT).
+		WithSequence(accI.GetSequence())
+	builder, err := txFactory.BuildUnsignedTx(msgs...)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := builder.SetMsgs(msgs...); err != nil {
+		return nil, err
+	}
+
+	// set params
+	builder.SetGasLimit(gasLimit)
+	builder.SetFeeAmount(sdk.NewCoins(sdk.NewCoin(sdk.DefaultBondDenom, math.NewInt(int64(fee)))))
```

Comment:
Might be useful to have the fee amount be a coin instead of a uint in the case where we want to do oracle fee swaps if the fee market supports it.